pypdf/_text_extraction/_layout_mode/_font.py

```diff
@@ -66,8 +68,9 @@ def __post_init__(self) -> None:
                         # warning and or use reader's "strict" to force an ex???
                         continue
                     # check for format (1): `int [int int int int ...]`
-                    if isinstance(_w[idx + 1], Sequence):
-                        start_idx, width_list = _w[idx : idx + 2]
+                    w_next_entry = _w[idx + 1].get_object()
+                    if isinstance(w_next_entry, Sequence):
+                        start_idx, width_list = w_entry, w_next_entry
```

Comment:
in line 78, you should  replace it with `range(cast(int, start_idx), cast(int, start_idx) + len(width_list), 1),`
this will fix your mypy issue without impact on performances

Reply:
Done, although for the record, this was not my code. I didn't touch it. I also think if we address the fact that these should *all* be ints, and not floats, it would just fix itself.